### ISSUE TITLE
`aichat --list-models` should not require a model to be set

### ISSUE THREAD

**Issue by erikbjoern** [sigoden/aichat#513]
**Describe the bug**
When running `aichat --list-models` without a model set in `config.yaml`, the output is `Error: No available model`

**To Reproduce**
Run `aichat --list-models`

**Expected behavior**
That available models are listed

**Environment (please complete the following information):**
- os version: macOS 14.4.1 (23E224)
- aichat version: 0.17.0
- terminal version: Mac Terminal Version 2.14 (453)

**Comment by sigoden** [comment]
I cannot produce the debug.

What does your current `config.yaml` look like? 

What does "without a model set in config.yaml" mean?

@erikbjoern

**Comment by sigoden** [comment]
https://github.com/sigoden/aichat/blob/1348a62e5f8bc140a7218fbfe1b73f990ab16101/src/config/mod.rs#L952-L954

This error is possible only if there are no clients in `config.yaml`. However, under normal use this is unlikely to happen.

**Comment by erikbjoern** [comment]
OK, I misunderstood the command. I thought it would list all possible models, regardless of current config. Sorry for incomplete steps to reproduce btw 😅

The situation was that I upgraded to `0.17.0` and at the same time to `gpt-4o` and got `Error: No model 'openai:gpt-4o'` and I thought it was because it was unsupported. So I wanted to list all options to make sure I spelled it correctly etc, but it kept saying 'Error No model...', and so I tried removing the model from the config altogether, resulting in `Error: No available model`

The real problem was that I hadn't updated my config to the new format:
```
clients:
- type: openai
  api_key: XXX
```


**So, the error message for this situation could be improved:**
Running `aichat` in `0.17.0` when config looks like this:
```
model: openai:gpt-3.5
api_key: XXX
```
the error is `Error: No model 'openai:gpt-3.5'`. It should rather say `LLM client/credentials not configured` or similar.
